Assignment/intlist.c

```diff
@@ -0,0 +1,129 @@
+   return list;
+}
+
+int Add (LinkedList* list, int val) {
+   Node* newNode = (Node*)malloc (sizeof (Node)); // Allocate memory for the new node
+   if (newNode == NULL) return ERROR_MEMORY_ALLOCATION;
+   newNode->data = val;
+   newNode->next = NULL;
+   if (list->head == NULL) list->head = newNode;
+   else {
+      Node* current = list->head;
+      while (current->next != NULL) {
+         current = current->next;
+      }
+      current->next = newNode;
+   }
+   return SUCCESS;
+}
+
+int Insert (LinkedList* list, int index, int val) {
+   if (index < 0) return ERROR_INVALID_INDEX;
+   Node* newNode = (Node*)malloc (sizeof (Node));
+   if (newNode == NULL) return ERROR_MEMORY_ALLOCATION;
+   newNode->data = val;
+   if (index == 0) {
+      newNode->next = list->head;
+      list->head = newNode; // Insert at the head of the list
+   }
+   else {
+      Node* current = list->head;
+      int i = 0;
+      while (current != NULL && i < index - 1) {
+         current = current->next;
+         i++;
+      }
+      if (current == NULL) {
+         free (newNode);
+         return ERROR_OUT_OF_BOUNDS;
+      }
+      newNode->next = current->next;
+      current->next = newNode;
+   }
+   return SUCCESS;
+}
+
+int RemoveAt (LinkedList* list, int index) {
+   if (index < 0) return ERROR_INVALID_INDEX;
+   Node* current = list->head;
+   Node* prev = NULL;
+   int i = 0;
+   while (current != NULL && i < index) {
+      prev = current;
+      current = current->next; // Traverse to the desired index
+      i++;
+   }
+   if (current == NULL) return ERROR_OUT_OF_BOUNDS;
+   if (prev == NULL) list->head = current->next;
+   else prev->next = current->next;
+   free (current);
+   return SUCCESS;
+}
+
+int Remove (LinkedList* list, int val) {
+   if (list->head == NULL) return ERROR_NOT_FOUND;
+   Node* current = list->head;
+   Node* prev = NULL;
+   if (current != NULL && current->data == val) {
+      list->head = current->next; // Remove the head node if it contains the value
+      free (current);
+      return SUCCESS;
+   }
+   // Traverse the list to find the first node with the given value
+   // If found, update the previous node's next pointer to bypass the node to be removed
+   while (current != NULL && current->data != val) {
+      prev = current;
+      current = current->next;
+   }
+   if (current == NULL) return ERROR_NOT_FOUND;
+   prev->next = current->next;
+   free (current);
+   return SUCCESS;
+}
+
+int Count (LinkedList* list) {
+   int count = 0;
+   Node* current = list->head;
+   while (current != NULL) {
+      count++;
+      current = current->next;
+   }
+   return count;
+}
+
+int Get (LinkedList* list, int index) {
+   Node* current = list->head;
+   for (int i = 0; i < index; i++) {
+      if (current == NULL) return ERROR_OUT_OF_BOUNDS;
+      current = current->next;
+   }
+   if (current == NULL) return ERROR_OUT_OF_BOUNDS;
+   return current->data;

```

Comment:
What if the value of the data matches the value of ERROR_OUT_OF_BOUNDS? How do you differentiate the error case from the actual value?